qiskit/transpiler/target.py

```diff
@@ -0,0 +1,652 @@
+        for the target"""
+        return [
+            (self._gate_name_map[op], qarg) for op in self._gate_map for qarg in self._gate_map[op]
+        ]
+
+    def instruction_properties(self, index):
+        """Get the instruction properties for a specific instruction tuple
+
+        This method is to be used in conjunction with the
+        :attr:`~qiskit.transpiler.Target.instructions` attribute of a
+        :class:`~qiskit.transpiler.Target` object. You can use this method to quickly
+        get the instruction properties for an element of
+        :attr:`~qiskit.transpiler.Target.instructions` by using the index in that list.
+        However, if you're not working with :attr:`~qiskit.transpiler.Target.instructions`
+        directly it is likely more efficient to access the target directly via the name
+        and qubits to get the instruction properties. For example, if
+        :attr:`~qiskit.transpiler.Target.instructions` returned::
+
+            [(XGate(), (0,)), (XGate(), (1,))]
+
+        you could get the properties of the ``XGate`` on qubit 1 with::
+
+            props = target.instruction_properties(1)
+
+        but just accessing it directly via the name would be more efficient::
+
+            props = target['x'][(1,)]
+
+        (assuming the ``XGate``'s canonical name in the target is ``'x'``)
+        This is especially true for larger targets as this will scale worse with the number
+        of instruction tuples in a target.
+
+        Args:
+            index (int): The index of the instruction tuple from the
+                :attr:`~qiskit.transpiler.Target.instructions` attribute. For, example
+                if you want the properties from the third element in
+                :attr:`~qiskit.transpiler.Target.instructions` you would set this to be ``2``.
+        Returns:
+            InstructionProperties: The instruction properties for the specified instruction tuple
+        """
+        instruction_properties = [
+            inst_props for op in self._gate_map for _, inst_props in self._gate_map[op].items()
+        ]
+        return instruction_properties[index]
+
+    def _build_coupling_graph(self):
+        self._coupling_graph = rx.PyDiGraph(multigraph=False)
+        self._coupling_graph.add_nodes_from([{} for _ in range(self.num_qubits)])
+        for gate, qarg_map in self._gate_map.items():
+            for qarg, properties in qarg_map.items():
+                if len(qarg) == 1:
+                    self._coupling_graph[qarg[0]] = properties
+                elif len(qarg) == 2:
+                    try:
+                        edge_data = self._coupling_graph.get_edge_data(*qarg)
+                        edge_data[gate] = properties
+                    except rx.NoEdgeBetweenNodes:
+                        self._coupling_graph.add_edge(*qarg, {gate: properties})
+
+    def coupling_map(self, two_q_gate=None):
```

Comment:
I would've expected this to be an `@property` (from consistency with the old `configuration`). Maybe `def build_coupling_map`, or leave `coupling_map` for compatibility only without the option to filter on gate name?

Reply:
It's a `@property` on the backend class to mirror the old `configuration` access patterns (although as I noted in the migration doc I wrote in #7323 the `BackendConfiguration` attribute returns an edge list not a `CouplingMap` object). It's a method here mostly because of the parameter, but also because of the return type and the overhead for construction. It felt more ergonomic to have it act as a function, because we're basically constructing a backwards compat object view from the target here and returning a separate object with a subset of the information contained in the target.

Reply:
That sounds fair. What do you think about giving it a verb name to match the fact that its a method and not a property (and reflect the build time cost)?

Reply:
sure, I'll rename it `build_coupling_map()`

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/24b60785140403a6589f53fc22f0c7520b35eb13